test/evp_kdf_test.c

```diff
@@ -160,6 +160,41 @@ static int test_kdf_pbkdf2(void)
     return ret;
 }
 
+#ifndef OPENSSL_NO_ARGON2
+static int test_kdf_argon2(void)
```

Comment:
Testing seems very light. In particular I would expect to see separate tests for all the new threading plumbing that this PR puts in.